src/navx_node.cpp

```diff
@@ -79,25 +158,21 @@ void update_callback(const ros::TimerEvent&) {
   msg.orientation.y = com->GetQuaternionY();
   msg.orientation.z = com->GetQuaternionZ();
   msg.orientation.w = com->GetQuaternionW();
-  msg.orientation_covariance[0] = 0;
-  msg.orientation_covariance[4] = 0;
-  msg.orientation_covariance[8] = 0;
   
-  msg.angular_velocity.x = com->GetPitchRate() * DEG_TO_RAD;
-  msg.angular_velocity.y = com->GetRollRate() * DEG_TO_RAD;
-  msg.angular_velocity.z = com->GetYawRate() * DEG_TO_RAD;
-  msg.angular_velocity_covariance[0] = 0;
-  msg.angular_velocity_covariance[4] = 0;
-  msg.angular_velocity_covariance[8] = 0;
+  msg.angular_velocity.x = curOrientation.ang_vel[0];
+  msg.angular_velocity.y = curOrientation.ang_vel[1];
+  msg.angular_velocity.z = curOrientation.ang_vel[2];
   
-  msg.linear_acceleration.x = com->GetWorldLinearAccelX() * 9.81;
-  msg.linear_acceleration.y = com->GetWorldLinearAccelY() * 9.81;
-  msg.linear_acceleration.z = com->GetWorldLinearAccelZ() * 9.81;
-  msg.linear_acceleration_covariance[0] = 0.8825985;
-  msg.linear_acceleration_covariance[4] = 0.8825985;
-  msg.linear_acceleration_covariance[8] = 1.569064;
-  
-  imu_pub.publish(msg);
+  msg.linear_acceleration.x = curOrientation.accel[0];
+  msg.linear_acceleration.y = curOrientation.accel[1];
+  msg.linear_acceleration.z = curOrientation.accel[2];
+
+  if (calculate_covariance(msg.orientation_covariance, 
+                           msg.angular_velocity_covariance, 
+                           msg.linear_acceleration_covariance)) {
```

Comment:
See comment by function prototypes above. This can become more legible by moving the inputs down a line, and only indenting once more. 
Where you put the closing/opening braces/curly braces are a matter of company convention, I forget what google's is, I think they'll stay on last line of final argument. Can change for readability, especially for weirder things.

Reply:
I also haven't mentioned, b/c I haven't noticed, using github which doesn't show these things, but a (nearly universal) rule on column widths are 80 (soft) max; different companies, for particularly bad lines, will have a hard max of like 92 or something. 
Also, as you start getting more experienced, and you start learning how to use your tools better, you'll start "window"-ing your files, splitting portions of your screen more & more efficiently to have more reference files and terminal buffers available all on the same monitor. Code that nicely fits into 40-50 characters, or that hugs the left margin, becomes more efficient to work with and use. For slightly similar and completely disastrous results, google the Arrow Anti-pattern. 
You can leave this as is, it is not worth my own time to bring it up twice.

Reply:
I unresolved this just so you could see the comment. It's resolved.